common/test/pacts/middleware.py

```diff
@@ -0,0 +1,33 @@
+"""
+Contain the middleware logic needed during pact verification
+"""
+
+from django.conf import settings
+from django.utils.deprecation import MiddlewareMixin
+
+from common.djangoapps.student.tests.factories import UserFactory
+
+
+class AuthenticationMiddleware(MiddlewareMixin):
+    """
+    Middleware to add default authentication into the requests for pact verification.
+
+    This middleware is required to add a default authenticated user and bypass CSRF validation
+    into the requests during the pact verification workflow. Without the authentication, the pact verification
+    process will not work as the apis.
+    See https://docs.pact.io/faq#how-do-i-test-oauth-or-other-security-headers
+    """
+    def __init__(self, get_response):
+        super().__init__()
+
+        username = getattr(settings, 'MOCK_USERNAME', 'Mock User')
```

Comment:
nit: add _ in username. Django usernames don't allow space

Reply:
As discussed, this username exists in contract.